pkg/types/rpm/rpm.proto

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.rekor.v2.types.rpm;
+
+import "google/api/field_behavior.proto";
+import "rekor_common.proto";
+
+option go_package = "github.com/sigstore/rekor/pkg/generated/protobuf/types/rpm";
+option java_package = "dev.sigstore.rekor.v2.types.rpm";
+option java_multiple_files = true;
+option java_outer_classname = "RekorRPMProto";
+
+/*
+ * ProposedRPMPackage represents the structure of the message required to present to Rekor
+ * to make an entry of the RPMPackage type.
+ */
+message ProposedRPMPackage {
+    /*
+     * Public key (currently only PGP supported)
+     */
+    dev.sigstore.rekor.v2.common.PublicKey public_key = 1 [(google.api.field_behavior) = REQUIRED];
+    /*
+     * RPM package which contains an embedded signature (this will not be persisted into the log)
+     */
+    bytes content                                     = 2 [(google.api.field_behavior) = REQUIRED];
+}
+
+/*
+ * RPMPackage represents the structure of the entry persisted in the transparency log for
+ * the RPMPackage type.
+ */
+message RPMPackage {
+    /*
+     * Digital signature and public key
+     */
+    dev.sigstore.rekor.v2.common.Signature signature = 1;
+    /*
+     * Hash over entire RPM package
+     */
+    dev.sigstore.rekor.v2.common.Hash hash           = 2;
+    /*
+     * Key / value pairs from RPM headers
+     */
+    map<string, string> headers                      = 3;
```

Comment:
Same question, does this allow arbitrary data?

Reply:
same as alpine, the code currently extracts and stores cryptographically-verified metadata from the package headers.